### ISSUE TITLE
Regression in System.Net.Primitives for UWP

### ISSUE THREAD

**Issue by CIPop** [dotnet/corefx#21233]
Running the UAP tests I'm getting the following for System.Net.Primitives:

```
0:017> !pe
Exception object: 0000018080001210
Exception type:   System.ExecutionEngineException
Message:          <none>
InnerException:   System.Diagnostics.Debug+DebugAssertException, Use !PrintException 0000018080638d88 to see more.
StackTrace (generated):
<none>
StackTraceString: <none>
HResult: 80131506
0:017> !PrintException /d 0000018080638d88
Exception object: 0000018080638d88
Exception type:   System.Diagnostics.Debug+DebugAssertException
Message:          We need to use an internal method named InternalSetName that is declared on Cookie.

   at System.Diagnostics.Debug.Assert(Boolean condition, String message, String detailMessage)
   at System.Net.CookieParser.get_InternalSetNameMethod()
   at System.Net.CookieParser.Get()
   at System.Net.CookieContainer.CookieCutter(Uri uri, String headerName, String setCookieHeader, Boolean isThrow)
   at System.Net.CookieContainer.SetCookies(Uri uri, String cookieHeader)
   at System.Net.Primitives.Functional.Tests.CookieContainerTest.<>c__DisplayClass29_0.<SetCookies_InvalidData_Throws>b__0()
   at Xunit.Assert.RecordException(Action testCode)
   at Xunit.Assert.Throws[T](Action testCode)
   at System.Net.Primitives.Functional.Tests.CookieContainerTest.SetCookies_InvalidData_Throws(Uri uri, String cookieHeader)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at Xunit.Sdk.TestInvoker`1.CallTestMethod(Object testClassInstance)
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<InvokeTestMethodAsync>b__1()
   at Xunit.Sdk.ExecutionTimer.<AggregateAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction)
   at Xunit.Sdk.ExceptionAggregator.<RunAsync>d__9.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code)
   at Xunit.Sdk.TestInvoker`1.<InvokeTestMethodAsync>d__46.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestInvoker`1.InvokeTestMethodAsync(Object testClassInstance)
   at Xunit.Sdk.TestInvoker`1.<<RunAsync>b__45_0>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestInvoker`1.<RunAsync>b__45_0()
   at Xunit.Sdk.ExceptionAggregator.<RunAsync>d__10`1.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.ExceptionAggregator.RunAsync[T](Func`1 code)
   at Xunit.Sdk.XunitTestRunner.<InvokeTestAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.XunitTestRunner.InvokeTestAsync(ExceptionAggregator aggregator)
   at Xunit.Sdk.ExceptionAggregator.<RunAsync>d__10`1.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.ExceptionAggregator.RunAsync[T](Func`1 code)
   at Xunit.Sdk.TestRunner`1.<RunAsync>d__43.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestRunner`1.RunAsync()
   at Xunit.Sdk.XunitTheoryTestCaseRunner.<RunTestAsync>d__9.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.XunitTheoryTestCaseRunner.RunTestAsync()
   at Xunit.Sdk.TestCaseRunner`1.<RunAsync>d__19.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestCaseRunner`1.RunAsync()
   at Xunit.Sdk.XunitTestMethodRunner.RunTestCaseAsync(IXunitTestCase testCase)
   at Xunit.Sdk.TestMethodRunner`1.<RunTestCasesAsync>d__32.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestMethodRunner`1.RunTestCasesAsync()
   at Xunit.Sdk.TestMethodRunner`1.<RunAsync>d__31.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestMethodRunner`1.RunAsync()
   at Xunit.Sdk.TestClassRunner`1.<RunTestMethodsAsync>d__38.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestClassRunner`1.RunTestMethodsAsync()
   at Xunit.Sdk.TestClassRunner`1.<RunAsync>d__37.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestClassRunner`1.RunAsync()
   at Xunit.Sdk.TestCollectionRunner`1.<RunTestClassesAsync>d__28.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestCollectionRunner`1.RunTestClassesAsync()
   at Xunit.Sdk.TestCollectionRunner`1.<RunAsync>d__27.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestCollectionRunner`1.RunAsync()
   at Xunit.Sdk.TestAssemblyRunner`1.<RunTestCollectionsAsync>d__42.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestAssemblyRunner`1.RunTestCollectionsAsync(IMessageBus messageBus, CancellationTokenSource cancellationTokenSource)
   at Xunit.Sdk.XunitTestAssemblyRunner.<RunTestCollectionsAsync>d__14.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.XunitTestAssemblyRunner.RunTestCollectionsAsync(IMessageBus messageBus, CancellationTokenSource cancellationTokenSource)
   at Xunit.Sdk.TestAssemblyRunner`1.<RunAsync>d__41.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.TestAssemblyRunner`1.RunAsync()
   at Xunit.Sdk.XunitTestFrameworkExecutor.<RunTestCases>d__6.MoveNext()
   at System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Xunit.Sdk.XunitTestFrameworkExecutor.RunTestCases(IEnumerable`1 testCases, IMessageSink executionMessageSink, ITestFrameworkExecutionOptions executionOptions)
   at XUnit.Runner.Uap.XunitTestRunner.<RunTests>d__2.MoveNext()
   at System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at XUnit.Runner.Uap.XunitTestRunner.RunTests(String arguments, StringBuilder sbLog, Action`1 uiLogger)
   at XUnit.Runner.Uap.MainPage.<>c__DisplayClass3_0.<RunXunitTestsInDirectory>b__0()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
InnerException:   <none>
StackTrace (generated):
<none>
StackTraceString: <none>
HResult: 80131500
```

**Comment by CIPop** [comment]
From the stack it is related to #19138.

**Comment by davidsh** [comment]
There should not have been an Debug.Assert for this.  This code was changed to add the "public" modifier to the "InternalSetName" method when compiling for UAP.  Then the reflection solution can be used even on ILC which normally prevents reflection.

This assert should be removed. Instead, there should be a future TODO to figure out how to publicly expose the needed method.

**Comment by tijoytom** [comment]
Should be fixed by #21280

**Comment by CIPop** [comment]
Thanks @tijoytom !
